miden-lib/asm/miden/tx.masm

```diff
@@ -65,27 +65,23 @@ end
 #! note_idx is the index of the crated note.
 export.create_note
     syscall.create_note
-    # => [note_idx, EMPTY_WORD, 0]
-
-    # clear the padding from the kernel response
-    movdn.4 dropw swap drop
-    # => [note_idx]
+    # => [note_idx, PAD(15)]
 end
 
 #! Adds the ASSET to the note specified by the index.
 #!
-#! Inputs: [note_idx, ASSET]
-#! Outputs: [note_idx]
+#! This procedure is expected to be invoked using a `call` instruction. It makes no guarantees about
+#! the contents of the `PAD` elements shown below. It is the caller's responsibility to make sure
+#! these elements do not contain any meaningful data.
+#!
+#! Inputs:  [ASSET, note_idx, PAD(11)]
+#! Outputs: [ASSET, note_idx, PAD(11)]
 #!
 #! note_idx is the index of the note to which the asset is added.
 #! ASSET can be a fungible or non-fungible asset.
 export.add_asset_to_note
```

Comment:
This procedure is invoked via `exec` instruction - not `call` instruction. So, the comment is not accurate here. And the signature for this procedure should probably be:
```
#! Inputs:  [ASSET, note_idx, ...]
#! Outputs: [ASSET, note_idx, ...]
```
Within this procedure, we should ensure that we pad and "unpad" the stack appropriately before making a `syscall`. That is, the stack before the `syscall` should look like this:
```
# => [ASSET, note_idx, ZERO(11)]
```
Unless, of course, we change the methodology to guarantee that `add_asset_to_note` in `api.masm` does not modify the `PAD` elements.